dd-sdk-android/src/test/kotlin/com/datadog/android/tracing/internal/utils/TracerExtensionsTest.kt

```diff
@@ -28,50 +22,37 @@ import org.mockito.quality.Strictness
 )
 
 @MockitoSettings(strictness = Strictness.LENIENT)
-@ForgeConfiguration(Configurator::class)
+@ForgeConfiguration(Configurator::class, seed = 0x4f36670aL)
 class TracerExtensionsTest {
 
+    lateinit var tracer: Tracer
+
+    @BeforeEach
+    fun `set up`() {
+        tracer = Tracer.Builder().build()
+    }
+
     @AfterEach
     fun `tear down`() {
-        GlobalTracer.get().setFieldValue("isRegistered", false)
-        GlobalTracer::class.java.setStaticValue("tracer", NoopTracerFactory.create())
+        tracer.scopeManager().active()?.close()
     }
 
     @Test
     fun `it will return the trace id and span id if there is an active span`(forge: Forge) {
-        // given
-        val config = DatadogConfig.Builder(forge.anAlphabeticalString()).build()
-        Datadog.initialize(mockContext(), config)
-        GlobalTracer.registerIfAbsent(Tracer.Builder().build())
 
         // when
-        val tracer = GlobalTracer.get()
         val span = tracer.buildSpan(forge.anAlphabeticalString()).start() as DDSpan
         tracer.activateSpan(span)
 
         // then
         assertThat(tracer.traceId()).isEqualTo(span.traceId.toString())
         assertThat(tracer.spanId()).isEqualTo(span.spanId.toString())
     }
-    @Test
-    fun `it will return null for trace and span id if there is no active span`(forge: Forge) {
```

Comment:
Was this failing ? maybe because there was an active span from other test where we forgot to close it. I think that's the fix you added in the "tearDown" but I think it will be ok to keep this test if you want I can open a PR after and fix this.